File: src/ExpoTtpEduModule.ts
import { requireNativeModule } from 'expo-modules-core';
import { ExpoTtpEduModule, ExpoTtpEduOptions } from './ExpoTtpEdu.types';

export default requireNativeModule<ExpoTtpEduModule>('ExpoTtpEdu');

export function showTapToPayEducation(options?: ExpoTtpEduOptions): Promise<void> {
  return requireNativeModule<ExpoTtpEduModule>('ExpoTtpEdu').showTapToPayEducation(options);
}

File: android/src/main/java/expo/modules/ttpedu/ExpoTtpEduModule.kt
package expo.modules.ttpedu

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition
import expo.modules.kotlin.Promise
import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class ExpoTtpEduOptions : Record {
  @Field
  val uiMode: String? = null
}

class ExpoTtpEduModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoTtpEdu")

    AsyncFunction("showTapToPayEducation") { options: ExpoTtpEduOptions? ->
      // Does nothing on Android
    }
  }
}
